editor/connections_dialog.cpp

```diff
@@ -1339,37 +1313,16 @@ void ConnectionsDock::update_tree() {
 	Ref<Script> script_base = selected_node->get_script();
 
 	while (native_base != StringName()) {
-		String class_name;
 		String doc_class_name;
-		String class_brief;
 		Ref<Texture2D> class_icon;
 		List<MethodInfo> class_signals;
 
 		if (script_base.is_valid()) {
-			class_name = script_base->get_global_name();
-			if (class_name.is_empty()) {
-				class_name = script_base->get_path().get_file();
-			}
-
 			doc_class_name = script_base->get_global_name();
 			if (doc_class_name.is_empty()) {
 				doc_class_name = script_base->get_path().trim_prefix("res://").quote();
-			}
-
-			// For a script class, the cache is filled each time.
-			if (!doc_class_name.is_empty()) {
-				if (descr_cache.has(doc_class_name)) {
-					descr_cache[doc_class_name].clear();
-				}
-				HashMap<String, DocData::ClassDoc>::ConstIterator F = doc_data->class_list.find(doc_class_name);
-				if (F) {
-					class_brief = F->value.brief_description;
-					for (int i = 0; i < F->value.signals.size(); i++) {
-						descr_cache[doc_class_name][F->value.signals[i].name] = F->value.signals[i].description;
-					}
-				} else {
-					doc_class_name = String();
-				}
+			} else if (!doc_data->class_list.find(doc_class_name)) {
+				doc_class_name = String();
```

Comment:
This part is not equal the old code. If the line above returns a non-empty value, previously we would check it against doc data and may set `doc_class_name` back to empty. But now this will never happen.

Though I'm not entirely sure which case is that in practice.

Reply:
`doc_class_name` is still set to empty if it can't be found, so I'm not sure what you mean.

Reply:
```suggestion
			}
			if (!doc_data->class_list.find(doc_class_name)) {
				doc_class_name = String();
```